pkg/cache/resource.go

```diff
@@ -76,3 +72,45 @@ func GetResourceName(xds proto.Message) string {
 		return ""
 	}
 }
+
+// GetResourceReferences returns the names for dependent resources (EDS cluster
+// names for CDS, RDS routes names for LDS).
+func GetResourceReferences(resources ...Resource) map[string]bool {
+	out := make(map[string]bool)
+	for _, res := range resources {
+		if res != nil {
```

Comment:
i would change these conditionals from

```go
if res != nil {}
```

to

```go
if res == nil { continue }
```

Specifically to reduce the depth of this branch and make it easier to read.

Reply:
Done